src/sortStudents.ts

```diff
@@ -1,16 +1,74 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'averageGrade',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desk';
 
+export function sortStudents(students: Student[],
+  sortBy: SortType, order: SortOrder): Student[] {
+  const result: Student[] = [...students];
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  function averageGrade(student: Student): number {
+    return student.grades.reduce((a: number, b: number) => (
+      a + b), 0) / student.grades.length;
```

Comment:
Писати типи в callback, зазвичай не потрібно, TS сам зрозуміє, що там є, якщо ти вказав тип масиву

```suggestion
    return student.grades.reduce((a, b) => a + b, 0) / student.grades.length;
```